server/src/main/java/org/opensearch/index/shard/RemoteStoreRefreshListener.java

```diff
@@ -402,6 +439,7 @@ private boolean isRefreshAfterCommitSafe() {
     void uploadMetadata(Collection<String> localSegmentsPostRefresh, SegmentInfos segmentInfos, ReplicationCheckpoint replicationCheckpoint)
         throws IOException {
         final long maxSeqNo = ((InternalEngine) indexShard.getEngine()).currentOngoingRefreshCheckpoint();
```

Comment:
@ashking94  
```suggestion
        final long maxSeqNo = Long.parseLong(segmentInfos.getUserData().get(SequenceNumbers.MAX_SEQ_NO));
```
Same as above as  it will be consistent with the segmentInfo we started for upload and it will stop any race condition.

Reply:
this is definitely buggy after this change. We should be caching the max seq No now in the beforeListener method.